### PR TITLE
Migrate to gh actions

### PR DESCRIPTION
**Pull request by joseph-flinn**
## Summary
Migrating the AppVeyor pipeline to two GitHub Action pipelines: one for building and signing the windows executables and the other for building everything else.

This does not include releasing Linux or MacOS assets.

## Files Changed
- .github/workflows/build.yml
- .github/workflows/release.yml
- scripts/make-versioninfo.ps1
- package.json
- sign.js

## ⚠️ Notes
The signing secrets below can be found in the DevOps Collection

Repo Secrets Needed:

- SIGNING_VAULT_URL
- SIGNING_CLIENT_ID
- SIGNING_CLIENT_SECRET
- SIGNING_TENANT_ID
- SIGNING_CERT_NAME

### CODE REVIEW

File: .github/scripts/macos/decrypt-secrets.ps1 [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿$rootPath = $env:GITHUB_WORKSPACE;
+
+$decryptSecretPath = $($rootPath + "/.github/scripts/decrypt-secret.ps1");
+
+Invoke-Expression "& `"$decryptSecretPath`" -filename bitwarden-desktop-key.p12.gpg"
+Invoke-Expression "& `"$decryptSecretPath`" -filename appstore-app-cert.p12.gpg"
+Invoke-Expression "& `"$decryptSecretPath`" -filename appstore-installer-cert.p12.gpg"
```
Comment (kspearrin) [inline .github/scripts/macos/decrypt-secrets.ps1:7]: these dont exist?

File: .github/scripts/macos/increment-version.ps1 [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿$rootPath = $env:GITHUB_WORKSPACE;
```
Comment (kspearrin) [inline .github/scripts/macos/increment-version.ps1:1]: not sure this is necessary given we are not submitting to app store

File: .github/scripts/macos/setup-keychain.ps1 [hunk 3]
```diff
@@ -0,0 +1,21 @@
+$homePath = Resolve-Path "~" | Select-Object -ExpandProperty Path;
+$secretsPath = $homePath + "/secrets"
+
+$desktopKeyPath = $($secretsPath + "/bitwarden-desktop-key.p12");
+$devidAppCertPath = $($secretsPath + "/devid-app-cert.p12");
+$devidInstallerCertPath = $($secretsPath + "/devid-installer-cert.p12");
+$appstoreAppCertPath = $($secretsPath + "/appstore-app-cert.p12");
```
Comment (kspearrin) [inline .github/scripts/macos/setup-keychain.ps1:7]: some of these not needed?

File: .github/scripts/macos/setup-profiles.ps1 [hunk 4]
```diff
@@ -0,0 +1,6 @@
+$homePath = Resolve-Path "~" | Select-Object -ExpandProperty Path;
+$secretsPath = $homePath + "/secrets"
+$rootPath = $env:GITHUB_WORKSPACE
+$pprofile = "bitwarden_desktop_appstore.provisionprofile"
```
Comment (kspearrin) [inline .github/scripts/macos/setup-profiles.ps1:4]: not needed since no app store

**Review by kspearrin — CHANGES_REQUESTED** [review]

**Comment by kspearrin** [comment]
i think you need to go through and remove all the stuff related to app stores since we don't submit DC there

**Comment by joseph-flinn** [comment]
Yep, will do. I think I just did a quick grab of all of the desktop mac stuff and put it in this repo before working on the mac portion. 

This PR doesn't have the mac portion enabled. It is here for if we want to release DC previous to the mac portion being done. I'm also good with waiting if that's what we want to do

**Review by kspearrin — APPROVED** [review]

**Pull request merged**